### PR TITLE
Get rid of underscore

### PR DESCRIPTION
**Pull request by BigYu**

### CODE REVIEW

File: src/utils/function.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const partialApply = (fn, args) => fn.bind(null, ...args);
+
+export const isFunction = func =>
+  func && Object.prototype.toString.call(func) === '[object Function]';
```
Comment (ljqx) [inline src/utils/function.js]: no need for `func &&`

File: src/utils/decorator.js [hunk 2]
```diff
@@ -1,60 +1,59 @@
...
   }
 
   props(props, propsExt/* , context */) {
-    return _.extend({}, props, propsExt);
+    return Object.assign({}, props, propsExt);
   }
 
   classes(classes, classesExt/* , context */) {
-    return _.union(classes, classesExt);
+    return [...classes, ...classesExt];
```
Comment (ljqx) [inline src/utils/decorator.js]: no the same, like this it won't dedup

File: src/utils/object.js [hunk 3]
```diff
@@ -1,15 +1,29 @@
+import { flatten } from './array';
+
 export const isObject = obj => Object.prototype.toString.call(obj) === '[object Object]';
 
-export const compactObject = (obj) => {
+const iterateeWithCondition = (
+  obj,
+  condition = () => true,
```
Comment (ljqx) [inline src/utils/object.js]: the default value is not used

File: src/utils/function.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+
+export const partial = (fn, ...args) => partialApply(fn, args);
```
Comment (ljqx) [inline src/utils/function.js]: the interface is different with underscore

File: src/utils/convert.js [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
+import { isArray } from './array';
 
 export function convert(converter, value) {
-  if (_.isArray(value)) {
-    return _.map(value, converter);
+  if (isArray(value)) {
+    return value.map(converter);
   }
-  if (_.isNull(value) || _.isUndefined(value)) {
+  if (!value) {
```
Comment (ljqx) [inline src/utils/convert.js]: empty string?

File: src/utils/object.js [hunk 6]
```diff
@@ -1,15 +1,29 @@
...
   }, {});
 };
+
+export const compactObject = obj => iterateeWithCondition(obj, (item, key) => !!item[key]);
+
+export const mapObject = (obj, iteratee) => iterateeWithCondition(obj, () => true, iteratee);
+
+export const pick =
+  (obj, ...keys) =>
+    iterateeWithCondition(obj, (item, key) => flatten(keys).indexOf(key) > -1);
```
Comment (haipeng-wang) [inline src/utils/object.js]: no need to run flatten(keys) every time

File: src/builtin-projections/defaults.js [hunk 7]
```diff
@@ -10,60 +9,54 @@ function normalize(config) {
...
-
-  return _.defaults({
+  const colgroup = Object.assign({}, { columns }, config.colgroup);
+  const tbody = Object.assign({}, { records }, config.tbody);
+  const colgroups = (config.colgroups || [{ key: 'colgroup-default' }]).map(cg =>
+    (cg.key === 'colgroup-default' ? Object.assign(cg, colgroup) : cg));
+  const tbodies = (config.tbodies || [{ key: 'tbody-default' }]).map(tb =>
+    (tb.key === 'tbody-default' ? Object.assign(tb, tbody) : tb));
+
+  return Object.assign({}, config, {
```
Comment (ljqx) [inline src/builtin-projections/defaults.js]: `Object.assign` is not supported in IE, so we expect user to have polyfill?

Comment (BigYu) [reply src/builtin-projections/defaults.js]: IE sucks!!!!!!!!!!!

File: src/utils/object.js (hunk 6 above)
Comment (haipeng-wang) [reply src/utils/object.js]: flatten(keys).reduce((memo, key) => {
  if (obj.hasOwnProperty(key)) {
    return Object.assign(memo, {[key]: obj[key]});
  }
  return memo;
}, {})

File: src/utils/decorator.js (hunk 2 above)
Comment (BigYu) [reply src/utils/decorator.js]: Won't fix. :P

File: src/utils/object.js (hunk 6 above)
Comment (BigYu) [reply src/utils/object.js]:
```javascript
export const pick = (obj, ...keys) => {
  const flattendKeys = flatten(keys);

  return iterateeWithCondition(obj, (item, key) =>
    (searchKeys => searchKeys.indexOf(key) > -1)(flattendKeys));
};
```

File: src/index.js [hunk 8]
```diff
@@ -1,3 +1,6 @@
 import { ProjectionGridCore } from './projection-grid-core';
+import * as _ from './utils';
 
 export default ProjectionGridCore;
+
+export { _ };
```
Comment (wewei) [inline src/index.js]: Why do we export "_"

File: src/utils/array.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+export const pluck = (arr, key) => arr.map(item => item[key]);
+
+export const isArray = arr => Array.isArray(arr);
+
+export const flatten = list => list.reduce((a, b) => a.concat(isArray(b) ? flatten(b) : b), []);
```
Comment (wewei) [inline src/utils/array.js]: we should use the `a.concat(...arrays)` for better performance

File: src/index.js (hunk 8 above)
Comment (BigYu) [reply src/index.js]: For the react/vue project.

**Pull request merged**